README.md
# Administrative boundaries of Belgium based on Open Data available at Statistics Belgium

The  **BelgiumMaps.StatBel** package contains maps with administrative boundaries (national, regions, provinces, districts, municipalities, statistical sectors, agglomerations (200m)) of Belgium extracted from Open Data at Statistics Belgium. Data is converted from Lambert CRS into the WGS84 coordinate reference system. 

This data is available in several objects:

- BE_ADMIN_SECTORS: SpatialPolygonsDataFrame with polygons and data at the level of the statistical sector
- BE_ADMIN_MUNTY: SpatialPolygonsDataFrame with polygons and data at the level of the municipality
- BE_ADMIN_DISTRICT: SpatialPolygonsDataFrame with polygons and data at the level of the district
- BE_ADMIN_PROVINCE: SpatialPolygonsDataFrame with polygons and data at the level of the province
- BE_ADMIN_REGION: SpatialPolygonsDataFrame with polygons and data at the level of the region
- BE_ADMIN_BELGIUM: SpatialPolygonsDataFrame with polygons and data at the level of the whole of Belgium
- BE_ADMIN_HIERARCHY: data.frame with administrative hierarchy of Belgium
- BE_ADMIN_AGGLOMERATIONS: SpatialPolygonsDataFrame with polygons and data at the level of an agglomeration (200m)

![](inst/img/belgiummaps_statbel.png)


## Installation

The package is available at datatailor.be

```
install.packages("BelgiumMaps.StatBel", repos = "http://www.datatailor.be/rcube", type = "source")
vignette("BelgiumMaps_AdministrativeAreas_StatBel", package = "BelgiumMaps.StatBel")
```

To install the latest version from the repository itself:

```
install.packages(c('sp', 'knitr', 'markdown'))
devtools::install_git("http://bnosac.all2all.org:10080/gitlab/open-source/BelgiumMaps.StatBel.git", build_vignettes = TRUE)
## See the vignette
vignette("BelgiumMaps_AdministrativeAreas_StatBel", package = "BelgiumMaps.StatBel")
```

## Support in geospatial analysis

Need support in geospatial analysis or geospatial mapping. 
Contact BNOSAC: http://www.bnosac.be



inst/NEWS
                CHANGES IN BelgiumMaps.StatBel VERSION 1.0

    o    Initial version of the package



R/datasets.R

#' @name BE_ADMIN
#' @aliases BE_ADMIN_SECTORS BE_ADMIN_MUNTY BE_ADMIN_DISTRICT BE_ADMIN_PROVINCE BE_ADMIN_REGION BE_ADMIN_BELGIUM
#' @title BE_ADMIN_SECTORS: Maps with administrative boundaries of Belgium extracted from Open Data at Statistics Belgium.
#' @description BE_ADMIN_SECTORS: Maps with administrative boundaries of Belgium extracted from Open Data at Statistics Belgium. Namely:
#' 
#' \itemize{
#' \item BE_ADMIN_SECTORS: SpatialPolygonsDataFrame with polygons and data at the level of the statistical sector
#' \item BE_ADMIN_MUNTY: SpatialPolygonsDataFrame with polygons and data at the level of the municipality
#' \item BE_ADMIN_DISTRICT: SpatialPolygonsDataFrame with polygons and data at the level of the district
#' \item BE_ADMIN_PROVINCE: SpatialPolygonsDataFrame with polygons and data at the level of the province
#' \item BE_ADMIN_REGION: SpatialPolygonsDataFrame with polygons and data at the level of the region
#' \item BE_ADMIN_BELGIUM: SpatialPolygonsDataFrame with polygons and data at the level of the whole of Belgium
#' }
#' The original data from Statistics Belgium contains the areas at the level of the statistical sectors. These were aggregated by using the
#' definition set forth on 20/01/2011 as available in the cs012011 field.
#' 
#' Mark that Brussels is not considered as a province but for convenience, the province level of Brussels is set to the region information.
#' More information about the data can be found in the inst/documentation folder.
#' 
#' The data contains the following elements which were available at different levels. 
#' \itemize{
#' \item objectid: id of the area
#' \item cs012011: statistical code of the sector at 20/01/2011
#' \item cs102001: statistical code of the sector at 20/10/2001
#' \item cs031991: statistical code of the sector at 19/03/1991
#' \item cs031981: statistical code of the sector at 19/03/1981
#' \item sector.nl: name of the statistical sector (dutch)
#' \item sector.fr: name of the statistical sector (french)
#' \item gemeente: name of the commune (dutch)
#' \item commune: name of the commune (french)
#' \item arrond.nl: name of the district (dutch)
#' \item arrond.fr: name of the district (french)
#' \item prov.nl: name of the province (dutch)
#' \item prov.fr: name of the province (french)
#' \item reg.nl: name of the region (dutch)
#' \item reg.fr: name of the region (french)
#' \item gis.perime: perimeter of the statistical sector
#' \item gis.area.h: surface of the statistical sector (hectare)
#' \item cad.area.h: surface of the statistical sector adjusted to the cadastral area of the commune (hectare)
#' \item nuts0: Eurostat NUTS code level 0
#' \item nuts1: Eurostat NUTS code level 1
#' \item nuts2: Eurostat NUTS code level 2
#' \item nuts3: Eurostat NUTS code level 3
#' \item gis.area.km2: surface of the statistical sector (square km)
#' \item cad.area.km2: surface of the statistical sector adjusted to the cadastral area of the commune (square km)
#' \item CD_REFNIS_SECTOR: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item CD_SECTOR: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_SECTOR_DESCR_NL: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_SECTOR_DESCR_FR: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item CD_MUNTY_REFNIS: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_MUNTY_DESCR_NL: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_MUNTY_DESCR_FR: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item CD_DSTR_REFNIS: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_ADM_DSTR_DESCR_NL: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_ADM_DSTR_DESCR_FR: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item CD_PROV_REFNIS: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_PROV_DESCR_NL: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_PROV_DESCR_FR: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item CD_RGN_REFNIS: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_RGN_DESCR_NL: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' \item TX_RGN_DESCR_FR: Information added based on cs012011 from the dataset BE_ADMIN_HIERARCHY
#' }
#' 
#' @docType data
#' @source \url{http://statbel.fgov.be/nl/binaries/scbel01012011_gen13_tcm325-275679.zip}
#' @references \url{http://statbel.fgov.be/nl/statistieken/opendata/datasets/tools}
#' @seealso \link{BE_ADMIN_HIERARCHY}
#' @examples
#' \dontrun{
#' data(BE_ADMIN_SECTORS)
#' class(BE_ADMIN_SECTORS)
#' str(BE_ADMIN_SECTORS@data)
#' 
#' library(sp)
#' plot(subset(BE_ADMIN_SECTORS, TX_RGN_DESCR_NL %in% "Brussels Hoofdstedelijk Gewest"))
#' data(BE_ADMIN_MUNTY)
#' data(BE_ADMIN_DISTRICT) 
#' data(BE_ADMIN_PROVINCE) 
#' data(BE_ADMIN_REGION) 
#' data(BE_ADMIN_BELGIUM) 
#' plot(BE_ADMIN_MUNTY)
#' plot(BE_ADMIN_DISTRICT)
#' plot(BE_ADMIN_PROVINCE, lwd = 2, add = TRUE)
#' }
NULL


#' @name BE_ADMIN_HIERARCHY
#' @title BE_ADMIN_HIERARCHY: Definitions of sectors/municipalities/districts/provinces/regions in Belgium.
#' @description BE_ADMIN_HIERARCHY: Definitions of sectors/municipalities/districts/provinces/regions in Belgium.
#' @docType data
#' @source \url{http://statbel.fgov.be/nl/binaries/TF_PSNL_INC_TAX_SECTOR_tcm325-278417.zip}
#' @references \url{http://statbel.fgov.be/nl/statistieken/opendata/datasets/werk}
#' @examples
#' data(BE_ADMIN_HIERARCHY)
#' str(BE_ADMIN_HIERARCHY)
NULL



#' @name BE_ADMIN_AGGLOMERATIONS
#' @title BE_ADMIN_AGGLOMERATIONS: Maps with administrative boundaries of agglomerations (200m) of Belgium extracted from Open Data at Statistics Belgium
#' @description BE_ADMIN_AGGLOMERATIONS: Maps with administrative boundaries of agglomerations (200m) of Belgium extracted from Open Data at Statistics Belgium.
#' Agglomerations (200m) are groups of buildings of which none of them are more than 200m away from the closest neighbouring building.
#' @docType data
#' @source \url{http://statbel.fgov.be/nl/binaries/SH_CENSUS_2011_AGGLOMERATIONS_200M.shp_tcm325-276234.zip}
#' @references \url{http://statbel.fgov.be/nl/statistieken/opendata/datasets/tools/}
#' @examples
#' \dontrun{
#' data(BE_ADMIN_AGGLOMERATIONS)
#' library(sp)
#' plot(BE_ADMIN_AGGLOMERATIONS)
#' }
NULL




inst/dev/09_example_visualisation.R
library(sp)
library(leaflet)
library(BelgiumMaps.StatBel)

## Get taxes / statistical sector
tempfile <- tempfile()
download.file("http://statbel.fgov.be/nl/binaries/TF_PSNL_INC_TAX_SECTOR_tcm325-278417.zip", tempfile)
unzip(tempfile, list = TRUE)
taxes <- read.table(unz(tempfile, filename = "TF_PSNL_INC_TAX_SECTOR.txt"), sep="|", header = TRUE, 
                    encoding = "UTF-8", stringsAsFactors = FALSE, quote = "", na.strings = c("", "C"))
colnames(taxes)[1] <- "CD_YEAR"

## Get taxes in last year
taxes <- subset(taxes, CD_YEAR == max(taxes$CD_YEAR))
taxes <- taxes[, c("CD_YEAR", "CD_REFNIS_SECTOR", 
                   "MS_NBR_NON_ZERO_INC", "MS_TOT_NET_TAXABLE_INC", "MS_AVG_TOT_NET_TAXABLE_INC", 
                   "MS_MEDIAN_NET_TAXABLE_INC", "MS_INT_QUART_DIFF", "MS_INT_QUART_COEFF", 
                   "MS_INT_QUART_ASSYM")]

## Join taxes with the map
data(BE_ADMIN_SECTORS, package = "BelgiumMaps.StatBel")
data(BE_ADMIN_DISTRICT, package = "BelgiumMaps.StatBel")
data(BE_ADMIN_MUNTY, package = "BelgiumMaps.StatBel")
mymap <- merge(BE_ADMIN_SECTORS, taxes, by = "CD_REFNIS_SECTOR", all.x=TRUE, all.y=FALSE)
#mymap <- subset(mymap, TX_RGN_DESCR_NL %in% "Brussels Hoofdstedelijk Gewest")

## Visualise the data
pal <- colorBin(palette = rev(heat.colors(11)), domain = mymap$MS_AVG_TOT_NET_TAXABLE_INC, 
                bins = c(0, round(quantile(mymap$MS_AVG_TOT_NET_TAXABLE_INC, na.rm=TRUE, probs = seq(0.1, 0.9, by = 0.1)), 0), +Inf),
                na.color = "#cecece")
m <- leaflet(mymap) %>%
  addTiles() %>%
  addLegend(title = "Net Taxable Income (EURO)", pal = pal, values = ~MS_AVG_TOT_NET_TAXABLE_INC, position = "bottomleft", na.label = "Missing") %>%
  addPolygons(color = ~pal(MS_AVG_TOT_NET_TAXABLE_INC), stroke = FALSE, smoothFactor = 0.2, fillOpacity = 0.85, 
              popup = sprintf("%s: %s<br>%s: %s<br><br>%s €: Average net taxable income<br>%s €: Median net taxable income<br>%s declarations", 
                              mymap$TX_SECTOR_DESCR_NL, mymap$TX_MUNTY_DESCR_NL, 
                              mymap$TX_SECTOR_DESCR_FR, mymap$TX_MUNTY_DESCR_FR,  
                              mymap$MS_AVG_TOT_NET_TAXABLE_INC, mymap$MS_MEDIAN_NET_TAXABLE_INC, 
                              mymap$MS_NBR_NON_ZERO_INC))
m <- addPolylines(m, data = BE_ADMIN_DISTRICT, weight = 1.5, color = "black")
m

## Save as HTML
library(htmlwidgets)
m$sizingPolicy$defaultHeight <- "100%"
m$sizingPolicy$defaultWidth <- "100%"
saveWidget(m, file="be_net_taxable_income_sector.html", selfcontained = FALSE)


inst/dev/01_import.R
library(rgdal)
library(sp)
library(rgeos)
library(tmap)
library(utils)
library(data.table)
library(maptools)

settings <- list()
settings$hierarchy <- list()
settings$hierarchy$municipalities <- file.path(getwd(), "inst", "extdata", "TF_PSNL_INC_TAX_MUNTY_tcm325-272513.zip")
settings$hierarchy$nis.sectors <- file.path(getwd(), "inst", "extdata", "TF_PSNL_INC_TAX_SECTOR_tcm325-278417.zip")
settings$nis.sectors <- file.path(getwd(), "inst", "extdata", "scbel01012011_gen13_tcm325-275679", "scbel01012011_gen13.shp")
settings$agglomeraties <- file.path(getwd(), "inst", "extdata", "SH_CENSUS_2011_AGGLOMERATIONS_200M.shp_tcm325-276234", "SH_CENSUS_2011_AGGLOMERATIONS_200M.shp")



##
## Import NIS hierarchy
##
unzip(settings$hierarchy$municipalities, list = TRUE)
unzip(settings$hierarchy$nis.sectors, list = TRUE)
hierarchy <- list()
hierarchy$municipalities <- read.table(unz(settings$hierarchy$municipalities, filename = "TF_PSNL_INC_TAX_MUNTY.txt"), sep="|", header = TRUE, 
                                       encoding = "UTF-8", stringsAsFactors = FALSE, quote = "")
hierarchy$nis.sectors <- read.table(unz(settings$hierarchy$nis.sectors, filename = "TF_PSNL_INC_TAX_SECTOR.txt"), sep="|", header = TRUE, 
                                       encoding = "UTF-8", stringsAsFactors = FALSE, quote = "")
hierarchy$municipalities <- unique(hierarchy$municipalities[, c("CD_MUNTY_REFNIS", "TX_MUNTY_DESCR_NL", "TX_MUNTY_DESCR_FR", 
                                                         "CD_DSTR_REFNIS", "TX_ADM_DSTR_DESCR_NL", "TX_ADM_DSTR_DESCR_FR", 
                                                         "CD_PROV_REFNIS", "TX_PROV_DESCR_NL", "TX_PROV_DESCR_FR", 
                                                         "CD_RGN_REFNIS", "TX_RGN_DESCR_NL", "TX_RGN_DESCR_FR")])
hierarchy$nis.sectors <- unique(hierarchy$nis.sectors[, c("CD_MUNTY_REFNIS", "CD_SECTOR", "CD_REFNIS_SECTOR", "TX_SECTOR_DESCR_NL", "TX_SECTOR_DESCR_FR")])
sum(duplicated(hierarchy$municipalities$CD_MUNTY_REFNIS)) ## doublecheck to make sure names did not change over the years
sum(duplicated(hierarchy$nis.sectors$CD_REFNIS_SECTOR)) ## doublecheck to make sure names did not change over the years
hierarchy <- merge(hierarchy$nis.sectors, hierarchy$municipalities, by = "CD_MUNTY_REFNIS")
BE_ADMIN_HIERARCHY <- hierarchy[, c("CD_REFNIS_SECTOR", "CD_SECTOR", "TX_SECTOR_DESCR_NL", "TX_SECTOR_DESCR_FR", 
                                    "CD_MUNTY_REFNIS", "TX_MUNTY_DESCR_NL", "TX_MUNTY_DESCR_FR", 
                                    "CD_DSTR_REFNIS", "TX_ADM_DSTR_DESCR_NL", "TX_ADM_DSTR_DESCR_FR", 
                                    "CD_PROV_REFNIS", "TX_PROV_DESCR_NL", "TX_PROV_DESCR_FR", 
                                    "CD_RGN_REFNIS", "TX_RGN_DESCR_NL", "TX_RGN_DESCR_FR")]

##
## Import shape of agglomerations
##
ogrListLayers(settings$agglomeraties)
ogrInfo(settings$agglomeraties, layer = "SH_CENSUS_2011_AGGLOMERATIONS_200M")

BE_ADMIN_AGGLOMERATIONS <- readShapeSpatial(fn = settings$agglomeraties, proj4string = CRS("+init=EPSG:31370"))
#BE_ADMIN_AGGLOMERATIONS <- readOGR(settings$agglomeraties, layer = "SH_CENSUS_2011_AGGLOMERATIONS_200M", stringsAsFactors = FALSE) ## will need to ask statbel why this is different
colnames(BE_ADMIN_AGGLOMERATIONS@data) <- tolower(make.names(colnames(BE_ADMIN_AGGLOMERATIONS@data), allow_=FALSE))
BE_ADMIN_AGGLOMERATIONS <- spTransform(x=BE_ADMIN_AGGLOMERATIONS, CRSobj = CRS("+proj=longlat +datum=WGS84"))

##
## Import shape of NIS sectors
##
ogrListLayers(settings$nis.sectors)
ogrInfo(settings$nis.sectors, layer = "scbel01012011_gen13")

BE_ADMIN_SECTORS <- readShapeSpatial(fn = settings$nis.sectors, proj4string = CRS("+init=EPSG:31370"))
#BE_ADMIN_SECTORS <- readOGR(settings$nis.sectors, layer = "scbel01012011_gen13", stringsAsFactors = FALSE) ## gives a shift when putting in WGS84
colnames(BE_ADMIN_SECTORS@data) <- tolower(make.names(colnames(BE_ADMIN_SECTORS@data), allow_=FALSE))

isutf8 <- function(x){
  for(column in colnames(x)){
    if(is.character(x[[column]]) | is.factor(x[[column]])){
      x[[column]] <- as.character(x[[column]])
      Encoding(x[[column]]) <- "UTF-8"
    }
  }
  x
}
BE_ADMIN_SECTORS@data <- isutf8(BE_ADMIN_SECTORS@data)
BE_ADMIN_SECTORS$nuts0 <- "BE"
BE_ADMIN_SECTORS@data <- setnames(BE_ADMIN_SECTORS@data, old = "nuts3.new", new = "nuts3")
BE_ADMIN_SECTORS@data <- BE_ADMIN_SECTORS@data[, c(setdiff(names(BE_ADMIN_SECTORS), c("nuts0", "nuts1", "nuts2", "nuts3")), c("nuts0", "nuts1", "nuts2", "nuts3"))]
## make brussels into a province
BE_ADMIN_SECTORS$prov.nl <- ifelse(is.na(BE_ADMIN_SECTORS$prov.nl), BE_ADMIN_SECTORS$reg.nl, BE_ADMIN_SECTORS$prov.nl)
BE_ADMIN_SECTORS$prov.fr <- ifelse(is.na(BE_ADMIN_SECTORS$prov.fr), BE_ADMIN_SECTORS$reg.fr, BE_ADMIN_SECTORS$prov.fr)
## add area in km2
BE_ADMIN_SECTORS$gis.area.km2 <- BE_ADMIN_SECTORS$gis.area.h / 100
BE_ADMIN_SECTORS$cad.area.km2 <- BE_ADMIN_SECTORS$cad.area.h / 100
## convert to WGS84 coordinates instead of Lambert
BE_ADMIN_SECTORS <- spTransform(x=BE_ADMIN_SECTORS, CRSobj = CRS("+proj=longlat +datum=WGS84"))

## remove redundant data (not needed as in sector)
BE_ADMIN_SECTORS$nis.012011 <- NULL
BE_ADMIN_SECTORS$sec012011 <- NULL

## add information of hierarchy based on NIS definition in 20/01/2011
## mark that BE_ADMIN_HIERARCHY contains sectors not in BE_ADMIN_SECTORS$cs012011 (TAXES not linked to sector or to sector in other period namely cs102001)
##   this can be checked with x <- subset(BE_ADMIN_HIERARCHY, !CD_REFNIS_SECTOR %in%  BE_ADMIN_SECTORS$cs012011)
## also sectors seem to exist in the BE_ADMIN_SECTORS which are not in the taxes data first fix these
x <- subset(BE_ADMIN_SECTORS@data, !cs012011 %in% BE_ADMIN_HIERARCHY$CD_REFNIS_SECTOR)
x$CD_REFNIS_SECTOR <- x$cs012011
x$CD_MUNTY_REFNIS <- substr(x$CD_REFNIS_SECTOR, 1, 5)
x$CD_SECTOR <- substr(x$CD_REFNIS_SECTOR, 6, nchar(x$CD_REFNIS_SECTOR))
x <- setnames(x, old = c("sector.nl", "sector.fr"), new = c("TX_SECTOR_DESCR_NL", "TX_SECTOR_DESCR_FR"))
x <- x[, c("CD_REFNIS_SECTOR", "CD_SECTOR", "TX_SECTOR_DESCR_NL", "TX_SECTOR_DESCR_FR", "CD_MUNTY_REFNIS")]
x <- merge(x, unique(BE_ADMIN_HIERARCHY[, c(#"CD_REFNIS_SECTOR", "CD_SECTOR", "TX_SECTOR_DESCR_NL", "TX_SECTOR_DESCR_FR", 
  "CD_MUNTY_REFNIS", "TX_MUNTY_DESCR_NL", "TX_MUNTY_DESCR_FR", 
  "CD_DSTR_REFNIS", "TX_ADM_DSTR_DESCR_NL", "TX_ADM_DSTR_DESCR_FR", 
  "CD_PROV_REFNIS", "TX_PROV_DESCR_NL", "TX_PROV_DESCR_FR", 
  "CD_RGN_REFNIS", "TX_RGN_DESCR_NL", "TX_RGN_DESCR_FR")]), by = "CD_MUNTY_REFNIS")
x <- x[, names(BE_ADMIN_HIERARCHY)]
BE_ADMIN_HIERARCHY <- rbind(BE_ADMIN_HIERARCHY, x) 
## mark that sectors can be split or aggregated from year to year (see subset(BE_ADMIN_SECTORS, cs102001 == "52063A372")@data) and Hmisc::describe(BE_ADMIN_SECTORS@data) (look at uniques in sector ids)
## there seems to be one element per cs012011 but doubles in older NIS versions. will use version cs012011
BE_ADMIN_SECTORS$CD_REFNIS_SECTOR <- BE_ADMIN_SECTORS$cs012011
BE_ADMIN_SECTORS <- merge(BE_ADMIN_SECTORS, BE_ADMIN_HIERARCHY, by.x = "CD_REFNIS_SECTOR", by.y = "CD_REFNIS_SECTOR", all.x=FALSE, all.y=FALSE)
BE_ADMIN_SECTORS@data <- BE_ADMIN_SECTORS@data[, c(setdiff(names(BE_ADMIN_SECTORS), colnames(BE_ADMIN_HIERARCHY)), colnames(BE_ADMIN_HIERARCHY))]

## add nuts information to hierarchy
BE_ADMIN_HIERARCHY <- merge(BE_ADMIN_HIERARCHY, 
                            unique(na.exclude(BE_ADMIN_SECTORS@data[, c("CD_MUNTY_REFNIS", "nuts0", "nuts1", "nuts2", "nuts3")])), 
                            by = "CD_MUNTY_REFNIS", all.x=TRUE, all.y=FALSE)
BE_ADMIN_HIERARCHY <- BE_ADMIN_HIERARCHY[, c("CD_REFNIS_SECTOR", "CD_SECTOR", "TX_SECTOR_DESCR_NL", "TX_SECTOR_DESCR_FR", 
                                             "CD_MUNTY_REFNIS", "TX_MUNTY_DESCR_NL", "TX_MUNTY_DESCR_FR", 
                                             "CD_DSTR_REFNIS", "TX_ADM_DSTR_DESCR_NL", "TX_ADM_DSTR_DESCR_FR", 
                                             "CD_PROV_REFNIS", "TX_PROV_DESCR_NL", "TX_PROV_DESCR_FR", 
                                             "CD_RGN_REFNIS", "TX_RGN_DESCR_NL", "TX_RGN_DESCR_FR", "nuts0", "nuts1", "nuts2", "nuts3")]

x <- split(BE_ADMIN_SECTORS, BE_ADMIN_SECTORS$CD_MUNTY_REFNIS)
x <- lapply(x, FUN=function(data){
  SpatialPolygonsDataFrame(gUnaryUnion(spgeom = data),
                           data = data.frame(CD_MUNTY_REFNIS = unique(data$CD_MUNTY_REFNIS),
                                             TX_MUNTY_DESCR_NL = unique(data$TX_MUNTY_DESCR_NL),
                                             TX_MUNTY_DESCR_FR = unique(data$TX_MUNTY_DESCR_FR),
                                             CD_DSTR_REFNIS = unique(data$CD_DSTR_REFNIS),
                                             TX_ADM_DSTR_DESCR_NL = unique(data$TX_ADM_DSTR_DESCR_NL),
                                             TX_ADM_DSTR_DESCR_FR = unique(data$TX_ADM_DSTR_DESCR_FR),
                                             CD_PROV_REFNIS = unique(data$CD_PROV_REFNIS),
                                             TX_PROV_DESCR_NL = unique(data$TX_PROV_DESCR_NL),
                                             TX_PROV_DESCR_FR = unique(data$TX_PROV_DESCR_FR),
                                             CD_RGN_REFNIS = unique(data$CD_RGN_REFNIS),
                                             TX_RGN_DESCR_NL = unique(data$TX_RGN_DESCR_NL),
                                             TX_RGN_DESCR_FR = unique(data$TX_RGN_DESCR_FR),
                                             nuts0 = paste(unique(data$nuts0), collapse=","),
                                             nuts1 = paste(unique(data$nuts1), collapse=","),
                                             nuts2 = paste(unique(data$nuts2), collapse=","),
                                             nuts3 = paste(unique(data$nuts3), collapse=","),
                                             SURFACE.GIS.h = sum(data@data$gis.area.h),
                                             SURFACE.CAD.h = sum(data@data$cad.area.h),
                                             SURFACE.GIS.km2 = sum(data@data$gis.area.km2),
                                             SURFACE.CAD.km2 = sum(data@data$cad.area.km2),
                                             stringsAsFactors=FALSE))
})
BE_ADMIN_MUNTY <- do.call(sbind, x)
plot(BE_ADMIN_MUNTY)

x <- split(BE_ADMIN_SECTORS, BE_ADMIN_SECTORS$CD_DSTR_REFNIS)
x <- lapply(x, FUN=function(data){
  SpatialPolygonsDataFrame(gUnaryUnion(spgeom = data),
                           data = data.frame(CD_DSTR_REFNIS = unique(data$CD_DSTR_REFNIS),
                                             TX_ADM_DSTR_DESCR_NL = unique(data$TX_ADM_DSTR_DESCR_NL),
                                             TX_ADM_DSTR_DESCR_FR = unique(data$TX_ADM_DSTR_DESCR_FR),
                                             CD_PROV_REFNIS = unique(data$CD_PROV_REFNIS),
                                             TX_PROV_DESCR_NL = unique(data$TX_PROV_DESCR_NL),
                                             TX_PROV_DESCR_FR = unique(data$TX_PROV_DESCR_FR),
                                             CD_RGN_REFNIS = unique(data$CD_RGN_REFNIS),
                                             TX_RGN_DESCR_NL = unique(data$TX_RGN_DESCR_NL),
                                             TX_RGN_DESCR_FR = unique(data$TX_RGN_DESCR_FR),
                                             nuts0 = paste(unique(data$nuts0), collapse=","),
                                             nuts1 = paste(unique(data$nuts1), collapse=","),
                                             nuts2 = paste(unique(data$nuts2), collapse=","),
                                             nuts3 = paste(unique(data$nuts3), collapse=","),
                                             SURFACE.GIS.h = sum(data@data$gis.area.h),
                                             SURFACE.CAD.h = sum(data@data$cad.area.h),
                                             SURFACE.GIS.km2 = sum(data@data$gis.area.km2),
                                             SURFACE.CAD.km2 = sum(data@data$cad.area.km2),
                                             stringsAsFactors=FALSE))
})
BE_ADMIN_DISTRICT <- do.call(sbind, x)
plot(BE_ADMIN_DISTRICT)

x <- split(BE_ADMIN_SECTORS, BE_ADMIN_SECTORS$CD_PROV_REFNIS)
x <- lapply(x, FUN=function(data){
  SpatialPolygonsDataFrame(gUnaryUnion(spgeom = data),
                           data = data.frame(CD_PROV_REFNIS = unique(data$CD_PROV_REFNIS),
                                             TX_PROV_DESCR_NL = unique(data$TX_PROV_DESCR_NL),
                                             TX_PROV_DESCR_FR = unique(data$TX_PROV_DESCR_FR),
                                             CD_RGN_REFNIS = unique(data$CD_RGN_REFNIS),
                                             TX_RGN_DESCR_NL = unique(data$TX_RGN_DESCR_NL),
                                             TX_RGN_DESCR_FR = unique(data$TX_RGN_DESCR_FR),
                                             nuts0 = paste(unique(data$nuts0), collapse=","),
                                             nuts1 = paste(unique(data$nuts1), collapse=","),
                                             nuts2 = paste(unique(data$nuts2), collapse=","),
                                             SURFACE.GIS.h = sum(data@data$gis.area.h),
                                             SURFACE.CAD.h = sum(data@data$cad.area.h),
                                             SURFACE.GIS.km2 = sum(data@data$gis.area.km2),
                                             SURFACE.CAD.km2 = sum(data@data$cad.area.km2),
                                             stringsAsFactors=FALSE))
})
BE_ADMIN_PROVINCE <- do.call(sbind, x)
plot(BE_ADMIN_PROVINCE)


x <- split(BE_ADMIN_SECTORS, BE_ADMIN_SECTORS$CD_RGN_REFNIS)
x <- lapply(x, FUN=function(data){
  SpatialPolygonsDataFrame(gUnaryUnion(spgeom = data),
                           data = data.frame(CD_RGN_REFNIS = unique(data$CD_RGN_REFNIS),
                                             TX_RGN_DESCR_NL = unique(data$TX_RGN_DESCR_NL),
                                             TX_RGN_DESCR_FR = unique(data$TX_RGN_DESCR_FR),
                                             nuts0 = paste(unique(data$nuts0), collapse=","),
                                             nuts1 = paste(unique(data$nuts1), collapse=","),
                                             SURFACE.GIS.h = sum(data@data$gis.area.h),
                                             SURFACE.CAD.h = sum(data@data$cad.area.h),
                                             SURFACE.GIS.km2 = sum(data@data$gis.area.km2),
                                             SURFACE.CAD.km2 = sum(data@data$cad.area.km2),
                                             stringsAsFactors=FALSE))
})
BE_ADMIN_REGION <- do.call(sbind, x)
plot(BE_ADMIN_REGION)

BE_ADMIN_BELGIUM <- SpatialPolygonsDataFrame(gUnaryUnion(spgeom = BE_ADMIN_SECTORS),
                                             data = data.frame(nuts0 = paste(unique(BE_ADMIN_SECTORS$nuts0), collapse=","),
                                                               SURFACE.GIS.h = sum(BE_ADMIN_SECTORS@data$gis.area.h),
                                                               SURFACE.CAD.h = sum(BE_ADMIN_SECTORS@data$cad.area.h),
                                                               SURFACE.GIS.km2 = sum(BE_ADMIN_SECTORS@data$gis.area.km2),
                                                               SURFACE.CAD.km2 = sum(BE_ADMIN_SECTORS@data$cad.area.km2)))


save(BE_ADMIN_HIERARCHY, file = file.path(getwd(), "data", "BE_ADMIN_HIERARCHY.RData"), compress = 'xz')
save(BE_ADMIN_SECTORS, file = file.path(getwd(), "data", "BE_ADMIN_SECTORS.RData"), compress = 'xz')
save(BE_ADMIN_MUNTY, file = file.path(getwd(), "data", "BE_ADMIN_MUNTY.RData"), compress = 'xz')
save(BE_ADMIN_DISTRICT, file = file.path(getwd(), "data", "BE_ADMIN_DISTRICT.RData"), compress = 'xz')
save(BE_ADMIN_PROVINCE, file = file.path(getwd(), "data", "BE_ADMIN_PROVINCE.RData"), compress = 'xz')
save(BE_ADMIN_REGION, file = file.path(getwd(), "data", "BE_ADMIN_REGION.RData"), compress = 'xz')
save(BE_ADMIN_BELGIUM, file = file.path(getwd(), "data", "BE_ADMIN_BELGIUM.RData"), compress = 'xz')
save(BE_ADMIN_AGGLOMERATIONS, file = file.path(getwd(), "data", "BE_ADMIN_AGGLOMERATIONS.RData"), compress = 'xz')


# library(rmapshaper)
# x <- ms_simplify(BE_ADMIN_REGION, keep = 0.8)
# plot(BE_ADMIN_BELGIUM)
# save(BE_ADMIN_SECTORS, file = "BE_ADMIN_SECTORS.RData", compress = 'xz')
# save(BE_ADMIN_MUNTY, file = "BE_ADMIN_MUNTY.RData", compress = 'xz')
# save(BE_ADMIN_DISTRICT, file = "BE_ADMIN_DISTRICT.RData", compress = 'xz')
# save(BE_ADMIN_PROVINCE, file = "BE_ADMIN_PROVINCE.RData", compress = 'xz')
# save(BE_ADMIN_REGION, file = "BE_ADMIN_REGION.RData", compress = 'xz')
# save(BE_ADMIN_BELGIUM, file = "BE_ADMIN_BELGIUM.RData", compress = 'xz')
# plot(x)
# x <- gSimplify(BE_ADMIN_BELGIUM, tol = 10)
# 
# Hmisc::describe(BE_ADMIN_SECTORS@data)
